plugins/volume-tagging/volume-tagger.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"fmt"
+	"github.com/antongulenko/golib"
+	collector "github.com/bitflow-stream/go-bitflow-collector"
+	"github.com/bitflow-stream/go-bitflow-collector/libvirt"
+	"github.com/bitflow-stream/go-bitflow/bitflow"
+	"github.com/bitflow-stream/go-bitflow/script/reg"
+	log "github.com/sirupsen/logrus"
+	"strings"
+	"sync"
+)
+
+func RegisterLibvirtVolumeTagger(name string, b reg.ProcessorRegistry) {
+	_ = b.RegisterStep(name, func(p *bitflow.SamplePipeline, params map[string]interface{}) error {
+		step := NewLibvirtVolumeTagger(params["uri"].(string), libvirt.NewDriver(), params["volumeKey"].(string),
+			params["libvirtInstanceKey"].(string))
+		p.Add(step)
+		return nil
+	}, "Append volume IDs to libvirt VM samples.").
+		Optional("uri", reg.String(), libvirt.LocalUri).
+		Optional("volumeKey", reg.String(), "volumes").
+		Optional("libvirtInstanceKey", reg.String(), "vm")
+	//steps.AddTagChangeListenerParams(step)
+}
+
+func NewLibvirtVolumeTagger(uri string, driver libvirt.Driver, volumeKey string, libvirtInstanceKey string) *LibvirtVolumeTagger {
+	return &LibvirtVolumeTagger{
+		connectUri:         uri,
+		driver:             driver,
+		volumeKey:          volumeKey,
+		libvirtInstanceKey: libvirtInstanceKey,
+	}
+}
+
+type LibvirtVolumeTagger struct {
+	bitflow.NoopProcessor
+	connectUri string
+	driver     libvirt.Driver
+	domains    map[string]libvirt.Domain
+
+	volumeKey          string
+	libvirtInstanceKey string
+}
+
+func (l *LibvirtVolumeTagger) Init() error {
+	l.domains = make(map[string]libvirt.Domain)
+	if err := l.fetchDomains(false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *LibvirtVolumeTagger) Start(wg *sync.WaitGroup) golib.StopChan {
+	if err := l.Init(); err != nil {
+		log.Warn("Error while initially updating libvirt domains.")
+	}
+	return l.NoopProcessor.Start(wg)
+}
+
+func (l *LibvirtVolumeTagger) fetchDomains(checkChange bool) error {
```

Comment:
Hmm too bad this duplicates with the libvirt data collection... :/ no easy way around this right now